lib/elixir/test/elixir/kernel/with_test.exs

```diff
@@ -48,6 +52,10 @@ defmodule Kernel.WithTest do
     assert_raise RuntimeError, fn ->
       with({:ok, res} <- ok(42), res = res + oops(), do: res)
     end
+
+    assert_raise WithClauseError, fn ->
```

Comment:
Please move it to the its own test as it not really related to the "errors in with" test (it is testing that raising propagates from `with`).